lib/hd/keyorigin.js

```diff
@@ -0,0 +1,168 @@
+'use strict';
+
+const assert = require('assert');
+const common = require('./common');
+const bio = require('bufio');
+const hash160 = require('bcrypto/lib/hash160');
+const WalletKey = require('../wallet/walletkey');
+
+/**
+ * Helper class to represent hd key path for arbitrary wallets.
+ * @property {Buffer} fingerPrint - master key finger print
+ * @property {Array} path - bip32 derivation path.
+ * @property {Number} path.index - index for derivation path.
+ */
+class KeyOriginInfo {
+  constructor(options) {
+    this.fingerPrint = -1;
+    this.path = [];
+    this.hardened = [];
```

Comment:
It looks like this member is unused and we rely on the path uint32 values to contain the hardened status internally.

Reply:
Thanks for review. Now fixed.
I'd better test against actual HW Wallets. But so far coldcard wallet is the only one supports PSBT as I know.
I'm now waiting for it to arrive. :(